### ISSUE TITLE
Missing default type when using functionBuilder

### ISSUE THREAD

**Issue by JBBianchi** [serverlessworkflow/sdk-typescript#90]
**What happened**:
`functionBuilder.build()`  doesn't assign a default type to the returned function.

**What you expected to happen**:
The returned function should have the type `rest` is none is supplied

**How to reproduce it**:
```typescript
const fn = functionBuilder().build();
console.log(fn.type);
```

**Anything else we need to know?**:
Will be fixed in an upcoming PR

**Comment by antmendoza** [comment]
Hi @JBBianchi 

I am fixing it right now as part os this issue #88

**Comment by JBBianchi** [comment]
Ok, I also have a fix for #89 , #90 and #91. I'll let you do the PR. Here are my changes to double check we are on the same page:

`tools/generate-builders.ts` #90 and #91
```typescript 
const buildersExtensions: { [key: string]: BuilderExtension } = {
...
  "Function": {
    preValidate: `\r\n    data.type =  data.type || 'rest';`
  },
  "Eventdef": {
    preValidate: `\r\n    data.kind =  data.kind || 'consumed';`
  },
};
```
`src/lib/builder.ts` #89 
```typescript 
export type Builder<T> = {
  build: () => T
} & {
  [k in T extends T ? keyof T : never]-?: (arg: T[k]) => Builder<T>
};
```
